Resources/doc/Sync/Relations/sql_relations.rst

```diff
@@ -0,0 +1,120 @@
+executing each relation as to mark down which changes in the database are relevant.
+
+    NOTE: Only ``DoctrineExtractor`` supports sql relations so far.
+
+Example extractor configuration:
+
+.. code-block:: yaml
+
+    # Extractor service.
+    ongr_connections.sync.extractor.doctrine_extractor:
+        class: ONGR\ConnectionsBundle\Sync\Extractor\DoctrineExtractor                    # Extractor class.
+        calls:
+            - [ setRelationsCollection, [ @ongr_connections.sync.relations_collection ] ] # Sql relation collection.
+            - [ setConnection, [ @database_connection ] ]                                 # Database collection.
+            - [ setStorageFacility, [ @ongr_connections.sync.sync_storage ] ]             # Sync storage provider.
+
+    # Relation collection service must be configured in order to use it.
+    ongr_connections.sync.relations_collection:
+        class: ONGR\ConnectionsBundle\Sync\Extractor\Relation\RelationsCollection
+
+
+Example simple sql relations setup:
+
+.. code-block:: yaml
+
+    services:
+        # Relation for category creation.
+        my_project.sql_relations.category.create:
+            class: %ongr_connections.sql_relations.composed_relation.class%
+            arguments: [my_categories, C, 1, category, NEW.categories_id]
+            tags:
+                - { name: ongr_connections.sql_relation }  # This tag is used to collect all sql relations.
+
+        # Relation for category update.
+        my_project.sql_relations.category.update:
+            class: %ongr_connections.sql_relations.composed_relation.class%
+            arguments: [my_categories, U, 1, category, NEW.categories_id]
+            tags:
+                - { name: ongr_connections.sql_relation }  # This tag is used to collect all sql relations.
+
+        # Relation for category deletion
+        my_project.sql_relations.category.delete:
+            class: %ongr_connections.sql_relations.composed_relation.class%
+            arguments: [my_categories, D, 1, category, OLD.categories_id]
+            tags:
+                - { name: ongr_connections.sql_relation }  # This tag is used to collect all sql relations.
+
+..
+
+Extractor constructor arguments are as follows:
+
+.. code-block:: php
+
+    /**
+     * @param string      $table        Table name to hook on.
+     * @param string      $type         Trigger and default job type C - create, U - update,  D - delete.
+     * @param int|null    $updateType   Partial update - 0, full update - 1.
+     * @param null        $documentType Type of target document.
+     * @param null        $idField      Source for document id.
+     * @param array       $trackFields  Array of table fields to track, all using default priority.
+     * @param string|null $jobType      C - create, U - update,  D - delete.
+     */
+..
+
+Last two parameters can and should be left undefined, as they are to be deprecated in near future.
```

Comment:
maybe we should leave out comments like "will be deprecated in near future", because we don't know yet, when that's gonna happen, lets just say that don't use them.